netsuite/client.py

```diff
@@ -83,8 +85,54 @@ def wrapper(self, *args, **kw):
     return decorator
 
 
+class NetSuiteTransport(Transport):
+    """
+    NetSuite dynamic domain wrapper for zeep.transports.transport
+
+    Latest NetSuite WSDL now uses relative definition addresses
+
+    zeep maps reflective remote calls to the base WSDL address,
+    rather than the dynamic subscriber domain
+
+    Wrap the zeep transports service with our address modifications
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Assign the dynamic host domain component to a class variable
+        """
+        wsdl_url = kwargs['wsdl_url']
+        kwargs.pop('wsdl_url',None)
+
+        parsed_wsdl_url = urlparse(wsdl_url)
+        self._wsdl_url = f'{parsed_wsdl_url.scheme}://{parsed_wsdl_url.netloc}/'
+
+        super().__init__(**kwargs)
```

Comment:
Could this maybe turn in to:

```python

    def __init__(self, wsdl_url, *args, **kwargs):
        """
        Assign the dynamic host domain component to a class variable
        """
        parsed_wsdl_url = urlparse(wsdl_url)
        self._wsdl_url = f'{parsed_wsdl_url.scheme}://{parsed_wsdl_url.netloc}/'

        super().__init__(*args, **kwargs)
```

It makes the init() requirement of wsdl_url more explicit and saves you the trouble of having to manipulate args/kwargs for the super(). (Generally, kwargs are optional arguments while positional args are required. It's also not very common in Python to pop kwargs like that.)

Then when you instantiate the class you remove the key and pass the URL as a positional argument.

```python
        return NetSuiteTransport(
            self._generate_wsdl_url(),
            session=self.session,
            cache=self.cache,
        )
```

Reply:
And for a deeper-dive into args/kwargs: https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3

Reply:
It seems like bad form to mix arguments by position, generic list, and key/value all in the same call.